src/tests/conformance/status.rs

```diff
@@ -0,0 +1,95 @@
+use std::{num::ParseIntError, time::Duration};
+
+use tempfile::TempDir;
+use tokio::time::{error::Elapsed, sleep, timeout};
+
+use crate::{
+    protocol::{codecs::binary::Payload, proto::TmStatusChange},
+    setup::node::build_stateful_builder,
+    tools::{
+        config::TestConfig,
+        constants::EXPECTED_RESULT_TIMEOUT,
+        rpc::{get_ledger_info, LedgerInfoResponse, RpcResponse},
+        synth_node::SyntheticNode,
+    },
+};
+
+#[tokio::test]
+#[allow(non_snake_case)]
+async fn C010_TM_STATUS_CHANGE_should_receive_ledger_information_by_status_change() {
+    let target = TempDir::new().expect("unable to create TempDir");
+
+    // Create a stateful node.
+    let mut node = build_stateful_builder(target.path().to_path_buf())
+        .expect("unable to get stateful builder")
+        .start(false)
+        .await
+        .expect("unable to start stateful node");
+
+    // Connect synth node.
+    let mut sn = SyntheticNode::new(&TestConfig::default()).await;
+    sn.connect(node.addr()).await.unwrap();
+
+    // Get ledger information via RPC.
+    let info = wait_for_ledger_info_by_rpc()
+        .await
+        .expect("no ledger info within required time");
+    let rpc_ledger_index = parse_ledger_index(&info).unwrap_or_else(|_| {
+        panic!(
+            "unable to parse ledger_index from response: '{}'",
+            info.result.ledger.ledger_index
+        );
+    });
+    let mut rpc_ledger_hash = [0u8; 32];
+    hex::decode_to_slice(&info.result.ledger.ledger_hash, &mut rpc_ledger_hash[..])
+        .expect("unable to decode ledger hash");
+
+    // Wait for TmStatusChange message.
+    wait_for_status_change(&mut sn, rpc_ledger_index, rpc_ledger_hash)
+        .await
+        .expect("no status message within required time");
+
+    // Cleanup.
+    node.stop().expect("unable to stop stateful node");
+    sn.shut_down().await;
```

Comment:
We should reverse the order here.

```suggestion
    sn.shut_down().await;
    node.stop().expect("unable to stop stateful node");
```